src/Bootstrap/src/Autoconfig/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,245 @@
+using Steeltoe.Management.Endpoint;
+using Steeltoe.Management.Kubernetes;
+using Steeltoe.Management.TaskCore;
+using Steeltoe.Management.Tracing;
+using Steeltoe.Security.Authentication.CloudFoundry;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+
+namespace Steeltoe.Bootstrap.Autoconfig
+{
+    public static class HostBuilderExtensions
+    {
+        private const string LoggerName = "Steeltoe.Autoconfig";
+        private static HashSet<string> missingAssemblies = new ();
+
+        static HostBuilderExtensions()
+        {
+            AppDomain currentDomain = AppDomain.CurrentDomain;
+            currentDomain.AssemblyResolve += LoadAnyVersion;
+            Assembly LoadAnyVersion(object sender, ResolveEventArgs args)
+            {
+                Assembly assembly = null;
+
+                // Load whatever version available - strip out version and culture info
+                string GetSimpleName(string assemblyName) => new Regex(",.*").Replace(assemblyName, string.Empty);
+                var name = GetSimpleName(args.Name);
+                if (missingAssemblies.Contains(name))
+                {
+                    return null;
+                }
+
+                var assemblies = AppDomain.CurrentDomain.GetAssemblies().ToDictionary(x => x.GetName().Name, x => x);
+                if (assemblies.TryGetValue(name, out assembly))
+                {
+                    return assembly;
+                }
+
+                if (args.Name?.Contains(".resources") ?? false)
+                {
+                    return args.RequestingAssembly;
+                }
+
+                missingAssemblies.Add(name); // throw it in there to prevent recursive attempts to resolve
+                assembly = Assembly.Load(name);
+                missingAssemblies.Remove(name);
+                return assembly;
+            }
+        }
+
+        public static IHostBuilder AddSteeltoe(this IHostBuilder hostBuilder, ILoggerFactory loggerFactory = null)
+        {
+            var logger = loggerFactory?.CreateLogger(LoggerName) ?? NullLogger.Instance;
+            hostBuilder.Properties[LoggerName] = logger;
+
+            if (!hostBuilder.WireIfLoaded(WireConfigServer, SteeltoeAssemblies.Steeltoe_Extensions_Configuration_ConfigServerCore))
+            {
+                hostBuilder.WireIfLoaded(WireCloudFoundryConfiguration, SteeltoeAssemblies.Steeltoe_Extensions_Configuration_CloudFoundryCore);
+            }
+
+            if (IsAssemblyLoaded(SteeltoeAssemblies.Steeltoe_Extensions_Configuration_KubernetesCore) && IsRunningInKubernetes())
+            {
+                WireKubernetesConfiguration(hostBuilder);
+            }
+
+            hostBuilder.WireIfLoaded(WirePlaceholderResolver, SteeltoeAssemblies.Steeltoe_Extensions_Configuration_PlaceholderCore);
+
```

Comment:
@hananiel should we autowire the SpringBoot config provider too? Which place in the order?

Reply:
Streams won't be included at this time, we can address the SpringBoot config provider later